pkg/rpcserver/runner.go

```diff
@@ -119,9 +191,15 @@ func (runner *Runner) connectionLoop() error {
 			return err
 		}
 	}
+
+	return nil
 }
 
 func (runner *Runner) sendRequest(req *queue.Request) error {
+	runner.mu.Lock()
```

Comment:
This lock is not necessary, right? The function can just use runner.conn as before.